### PR TITLE
 [TwigBridge] Remove $rootDir argument in CodeExtension

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | https://github.com/symfony/symfony-docs/pull/10547#issuecomment-432608538
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

Forgotten in #28890 

cc @javiereguiluz

### CODE REVIEW

File: src/Symfony/Bridge/Twig/Extension/CodeExtension.php [hunk 1]
```diff
@@ -23,21 +23,19 @@
...
     private $projectDir;
 
     /**
      * @param string|FileLinkFormatter $fileLinkFormat The format for links to source files
-     * @param string                   $rootDir        The project root directory
+     * @param string                   $projectDir     The project directory
      * @param string                   $charset        The charset
      */
-    public function __construct($fileLinkFormat, string $rootDir, string $charset, string $projectDir = null)
+    public function __construct($fileLinkFormat, string $projectDir, string $charset)
```
Comment (ro0NL) [inline src/Symfony/Bridge/Twig/Extension/CodeExtension.php]: argument was added in 4.2, so safe to remove technically.

So we slightly change the behavior, only in formatFile. IMHO that's fine.

Comment (ro0NL) [reply src/Symfony/Bridge/Twig/Extension/CodeExtension.php]: actually, we might assume the values will the same as we applied dirname() previously.

**Comment by nicolas-grekas** [comment]
can you have a look at the failure on Windows?
```
-'some/File.txt'
+'/some/File.txt'
C:\projects\symfony\src\Symfony\Bridge\Twig\Tests\Extension\CodeExtensionTest.php:28
```

**Comment by nicolas-grekas** [comment]
and the failure on Travis btw - looks like the composer.json change allowed spotting an issue :)

**Comment by ro0NL** [comment]
Im not sure the appveyor is still related, looks like it's not.

**Comment by ro0NL** [comment]
The `CodeHelper` looks like

https://github.com/symfony/symfony/blob/8c24c35fe86efbdb5f5d74c58d89518cb2865ea8/src/Symfony/Bundle/FrameworkBundle/Templating/Helper/CodeHelper.php#L33-L38

so i tend to keep the slash normalization actually =/

**Review by Tobion — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @ro0NL.

**Pull request merged**